ledger/sandbox-on-x/reference.conf

```diff
@@ -433,6 +433,83 @@ ledger {
 
         # Timeout for package metadata preparation after which a warning will be logged
         prepare-package-metadata-time-out-warning = 1 second
+
+        # Flat transactions stream configuration.
+        #
+        # A request to stream flat transactions contains a filtering constraint
+        # containing templates and parties that determine which transactions to stream.
+        # That filtering constraint is decomposed into simple filtering constraints each specifying at most one party and template.
+        # Each simple filtering constraint corresponds to a separate event id fetching query and an id page buffer.
+        # The fetched ids are then used to retrieve event payloads.
+        # Event payloads are then grouped in transactions and processed.
+        transactions-flat-streams {
+            # Upper bound on the number of event ids to retrieve in a single event id page (i.e. in a single event id fetch query).
+            max-ids-per-id-page=20000
+            # Upper bound on the number of id pages to fetch and store in a per-simple-filtering-constraint buffer before back-pressuring.
+            max-pages-per-id-pages-buffer=1
+            # Upper bound on the number of per-stream parallel id fetching queries for consuming events.
+            max-parallel-id-consuming-queries=4
+            # Upper bound on the number of per-stream parallel id fetching queries for create events.
+            max-parallel-id-create-queries=4
+            # Upper bound on the number of per-stream parallel payload fetching queries for consuming events. Must be a power of 2.
```

Comment:
I've seen "must be a power of 2" in multiple places in this PR, but I actually don't understand why it should be a power of 2...

Reply:
It's related to how we build our Akka streams. If you used a wrong value you would get
```
    //           java.lang.IllegalArgumentException: buffer size must be a power of two
    //	         at akka.stream.impl.fusing.ActorGraphInterpreter$BatchingActorInputBoundary.<init>(ActorGraphInterpreter.scala:128)
```
from here
https://github.com/digital-asset/daml/blob/main/ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionsFlatStreamReader.scala#L161-L167
Also, afair, without the preceding `.async` you can size the buffer two values not restricted to powers of 2


I wasn't able to find any Akka documentation on this error.